### DISCUSSION TITLE
PowerBI conversion to Reveal

### DISCUSSION THREAD

**Discussion by Codewow2** [RevealBi/Reveal.Sdk#274] [General]
Hi all,

I am tasked with converting as much functionality from PowerBI over to Reveal to maintain direct integrations with an existing system that implements Reveal.

I am worried about converting over some things like slicers since it looks like Reveal requires the dataset to be the filter's data provider specifically, rather than having a dataset of filter values separately that can be linked across multiple datasets.

Is there any way to make this kind of filtering an option?

To get even deeper into the weeds. Is it also possible to make these filters that affect visualizations pull the data they're looking for as a sort of live query rather than needing to obtain all of the data upfront?

**Comment by MLopezIG** [comment]
Hi there! I'll be assuming you're referring to dashboard filters, and it looks like you're under the impression that the only choice is to use the same 'dataset' (we call it data source/item) as the one being used in the dashboard. That's actually not the case, you can pick a different data source by choosing the following option: 

![image](https://github.com/user-attachments/assets/6f6f688b-8e2f-4689-b495-9d65c2422138)

I hope this helps. If my assumptions were wrong, please give some more information on what you want to achieve (maybe some screenshot will help).

Regarding how those filters pull data, they're limited to ~3000 results. If you need something that is not within those limits, the search functionality should help the user to find what it needs. Not sure if this falls under your 'live query' requirement, but it is not loading all the data upfront.

**Reply by Codewow2** [reply]
Appreciate the response! The filter/data source piece does help answer that!

I am running into filter trouble though. Here is what I am looking at at the moment. 
I have my list of filter options and I am trying to filter the list with search and it doesn't seem to do anything other than refresh the whole list of the first ~3000 records. Is this supposed to be making the list smaller?



![image](https://github.com/user-attachments/assets/4fba77eb-158f-4fed-ae93-9b4bb2b7eec3)

Actually, testing it on a list much smaller (only 12 records) seems to filter properly, so is this more of a bug or a limitation in the search filter?
![image](https://github.com/user-attachments/assets/813587a6-433b-4bfe-87ac-e8985292be69)


Thanks again! Are there any community resources outside of github that I can pop in to and ask questions as well? I'll be working heavily with Reveal for the foreseeable future and I know plenty of other questions will come up along the way.

**Reply by brianlagunas** [reply]
That does look like a bug. You should report that if you can duplicate it consistently. 

Contact your sales rep for a link to our Discord server. That is the best place to ask questions.

**Reply by Codewow2** [reply]
Thanks Brian, just waiting on a reply from our rep now! I'll see if it happens more as I go and submit an issue!
